### PR TITLE
Support Python 3 (work in progress)

### PR DESCRIPTION
**Pull request by mivade**
*This is a work in progress*. Several tests fail at present, mostly related to confusion between `str` and `bytes`. I will go through test failures and fix the obvious problems and mark the less obvious problems.

### CODE REVIEW

File: lib/hdf5db.py [hunk 1]
```diff
@@ -1324,11 +1328,10 @@ def makeNullTermStringAttribute(self, obj, attr_name, strLength, value):
         tid.set_size(strLength)
         tid.set_strpad(h5py.h5t.STR_NULLTERM)
         sid = h5py.h5s.create(h5py.h5s.SCALAR)
-        aid = h5py.h5a.create(obj.id, attr_name, tid, sid)
```
Comment (jreadey) [inline lib/hdf5db.py]: Not setting aid here?

**Comment by jreadey** [comment]
Nice code cleanup!  I made one comment in the file change listing - can you look at it?

The failures all seem related to attributes.  Are there any other types?

File: lib/hdf5db.py (hunk 1 above)
Comment (mivade) [inline lib/hdf5db.py]: Reviewing that change, it appears to have been the result of overzealous linting. Good catch.

**Comment by mivade** [comment]
The line change is fixed. It was related to accidentally deleting a `ret = ...` statement for which `ret` was not actually used. This then made the linter complain about assigning `aid` but never using it (which of course I don't use it if I delete that line!). I have added the assignment of `aid` back but removed assigning `ret` (still doing the `aid.write` call, however).

In this second take, I haven't seen any other errors yet. What I noticed on the first try was that getting some of these tests to pass resulted in further errors down the line. It would be nice to streamline the testing a bit, but I think that should wait until the remaining issues here can be resolved.

**Comment by jreadey** [comment]
Merging since everything checks out with py27.

**Pull request merged**